src/Microsoft.Data.Sqlite/SqliteConnection.cs

```diff
@@ -103,89 +104,103 @@ public override void Open()
+                        flags |= SQLITE_OPEN_URI;
+                        filename = "file:" + filename;
+                    }
+                    break;
+
+                default:
+                    Debug.Assert(
+                        ConnectionStringBuilder.Mode == SqliteOpenMode.ReadWriteCreate,
+                        "ConnectionStringBuilder.Mode is not ReadWriteCreate");
+                    flags |= SQLITE_OPEN_READWRITE | SQLITE_OPEN_CREATE;
+                    break;
             }
-        }
-#else
-        private readonly static string DefaultBasePath = PlatformServices.Default.Application.ApplicationBasePath;
-#endif
 
-        internal static string AdjustForRelativeDirectory(string path)
-        {
-            if (DefaultBasePath == null)
+            switch (ConnectionStringBuilder.Cache)
             {
-                return path;
+                case SqliteCacheMode.Shared:
+                    flags |= SQLITE_OPEN_SHAREDCACHE;
+                    break;
+
+                case SqliteCacheMode.Private:
+                    flags |= SQLITE_OPEN_PRIVATECACHE;
+                    break;
+
+                default:
+                    Debug.Assert(
+                        ConnectionStringBuilder.Cache == SqliteCacheMode.Default,
+                        "ConnectionStringBuilder.Cache is not Unspecified.");
+                    break;
             }
 
-            if (path.StartsWith("file:"))
+            if ((flags & SQLITE_OPEN_URI) == 0
+                && !filename.Equals(":memory:", StringComparison.OrdinalIgnoreCase)
+                && !Path.IsPathRooted(filename))
             {
-#if NETCORE50
-                // UWP cannot adjust for URI on UWP
-                return path;
-#else
-                // is this already absolute?
-                if (path.StartsWith("file:/"))
-                {
-                    return path;
-                }
-                var prefix = PlatformServices.Default.Runtime.OperatingSystem.Equals("Windows", StringComparison.OrdinalIgnoreCase) ? "file:///" : "file://";
-                return prefix + DefaultBasePath + "/" + path.Replace("file:", "");
-#endif
+                filename = Path.GetFullPath(Path.Combine(BaseDirectory, filename));
             }
 
-            if (Path.IsPathRooted(path))
-            {
-                return path;
-            }
+            var rc = NativeMethods.sqlite3_open_v2(filename, out _db, flags, vfs: null);
+            MarshalEx.ThrowExceptionForRC(rc, _db);
 
-            try
-            {
-                return Path.GetFullPath(Path.Combine(DefaultBasePath, path));
-            }
-            catch (Exception ex) when (ex is ArgumentException || ex is NotSupportedException)
+            SetState(ConnectionState.Open);
+
+            OnOpened();
+        }
+
+        partial void OnOpened();
+
+#if NETCORE50
+        partial void OnOpened()
+        {
+            var temporaryFolder = CurrentApplicationData?.TemporaryFolder.Path;
```

Comment:
Hmm... They're slightly different directories. The SQLite documentation uses `ApplicationData.Current.TemporaryFolder.Path`. We'll stick with that for now.